### PR TITLE
9.3 Multiple taxonomy rectors

### PR DESCRIPTION
**Pull request by bbrala**
## Description
Made a few rectors for: https://www.drupal.org/node/3039041 (also found here https://github.com/mglaman/drupal-change-record-triage/issues/282)

When the FunctionToStaticCallRector is merged, we can remove one of the implementation from this one ni favor of a confgiured rule.

## To Test
Unit and functional tests

## Drupal.org issue
Provide a link to the issue from https://www.drupal.org/project/rector/issues. If no issue exists, please create one and link to this PR.

### CODE REVIEW

**Comment by bbrala** [comment]
There are 2 failures left, but im not sure we can actually find those. Those assume we know something is a `Term` instance, which seems kinda optimistic?

**Comment by bbrala** [comment]
Implemented the optimistic approach, since the code would have been broken already if it was not a term.

**Comment by bbrala** [comment]
This needs to wait for deprecated fixes, since this will need to implement the versioned configuration.

File: config/drupal-9/drupal-9.3-deprecations.php [hunk 1]
```diff
@@ -41,4 +50,21 @@
...
+    // Missing: $url = $term->toUrl(); AND $name = taxonomy_term_title($term); AND taxonomy_implode_tags();
+    $rectorConfig->ruleWithConfiguration(FunctionToEntityTypeStorageMethod::class, [
+        new FunctionToEntityTypeStorageConfiguration('taxonomy_terms_static_reset', 'taxonomy_term', 'resetCache'),
+        new FunctionToEntityTypeStorageConfiguration('taxonomy_vocabulary_static_reset', 'taxonomy_vocabulary', 'resetCache'),
+    ]);
+    $rectorConfig->rule(TaxonomyVocabularyGetNamesRector::class);
+    $rectorConfig->rule(TaxonomyTermLoadMultipleByNameRector::class);
+    $rectorConfig->rule(TaxonomyVocabularyGetNamesDrupalStaticResetRector::class);
+    $rectorConfig->ruleWithConfiguration(FunctionToStaticRector::class, [
+        new FunctionToStaticConfiguration('taxonomy_implode_tags', 'Drupal\Core\Entity\Element\EntityAutocomplete', 'getEntityLabels'),
```
Comment (agentrickard) [inline config/drupal-9/drupal-9.3-deprecations.php]: New class params 
```    
    /**
     * @param string $introducedVersion Introduced version of this change
     * @param string $deprecatedFunctionName Deprecated function name
     * @param string $className Class to call static method on
     * @param string $methodName Method to call statically
     * @param array|int[] $argumentReorder Reorder arguments array[old_position] = new_position
     */
```

**Review by agentrickard — CHANGES_REQUESTED** [review]
Needs a minor update

**Review by agentrickard — APPROVED** [review]

**Pull request merged**